crates/uv-resolver/src/resolver/urls.rs

```diff
@@ -41,27 +56,77 @@ impl Urls {
         for requirement in manifest.requirements(markers, dependencies) {
             match &requirement.source {
                 RequirementSource::Registry { .. } => {}
-                RequirementSource::Url { url, .. } | RequirementSource::Path { url, .. } => {
+                RequirementSource::Url {
+                    subdirectory,
+                    location,
+                    url,
+                } => {
+                    let url = VerbatimParsedUrl {
+                        parsed_url: ParsedUrl::Archive(ParsedArchiveUrl {
+                            url: location.clone(),
+                            subdirectory: subdirectory.clone(),
+                        }),
+                        verbatim: url.clone(),
+                    };
                     if let Some(previous) = urls.insert(requirement.name.clone(), url.clone()) {
-                        if !is_equal(&previous, url) {
+                        // TODO(merge): use proper check
```

Comment:
What's this?

Reply:
At some future point in time, we should use an equality check by parts